gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1480,15 +1490,24 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 					dt = dt.Elem()
 					n.X = &StarExpr{X: n.X}
 					n.X.SetAttribute(ATTR_PREPROCESSED, true)
+					n.Addressability = AddressabilitySatisfied
 				}
 				switch dt.Kind() {
 				case StringKind, ArrayKind, SliceKind:
 					// Replace const index with int *ConstExpr,
 					// or if not const, assert integer type..
 					checkOrConvertIntegerKind(store, last, n.Index)
+					if dt.Kind() == SliceKind {
+						// A string index is not addressable.
```

Comment:
Confused by the comment

Also, maybe add a comment to explain what happens when dt is an array?

Reply:
Thanks. This comment is wrong. Updated. https://github.com/gnolang/gno/pull/2731/commits/548af8704c5227d8e8a077200d6c190950c12cd3

Reply:
If it is an array, it will defer to the array's addressability. Comment added. https://github.com/gnolang/gno/pull/2731/commits/26e8529c67cd846318c354ed8af1e4ca2bd25e9b